### ISSUE TITLE
Use progress package for auto progress bars

### ISSUE THREAD

**Issue by jefferis** [natverse/nat#275]
progress looks like exactly what I needed, because they don't turn on until after a small delay.

```
full_kcs.dps=dotprops(full_kcs/1e3, resample=1, k=5, .progress='progress')
```

Currently this shows two progress bars – one for the division (instantaneous) and one for the resample/dotprops operation.

See https://github.com/gaborcsardi/progress/issues/30

**Comment by jefferis** [comment]
made a start on this, but think I will hold off until issue https://github.com/gaborcsardi/progress/issues/33 is addressed i.e. show_after stops progress bars from being shown at all. That said progress is already cleaner than dply progress bars since they are cleared after use and they leave no footprint in the terminal (and one blank line in Rstudio).

**Comment by jefferis** [comment]
Main things left to do

* decide how to implement a default state for `nat.progress`.
* replace any other uses of progress bars that do not use nlapply

**Comment by jefferis** [comment]
On branch https://github.com/jefferis/nat/tree/feature/read.neurons-progress

**Comment by jefferis** [comment]
Also can give nmapply a progress bar

**Comment by jefferis** [comment]
closed by b2c17b3aeefd9831b4b3815c4f3127e05e1ff93c
